PlaceRouteHierFlow/cap_placer/capplacer.cpp

```diff
@@ -1770,8 +1770,8 @@ void Placer_Router_Cap::Common_centroid_capacitor_aspect_ratio(const string& opa
 
   for(unsigned int i = 0;i<current_node.Blocks.size();i++){
 
-      auto& b = current_node.Blocks[i].instance.back();
-      //PnRDB::block b = current_node.Blocks[i].instance[current_node.Blocks[i].instance.size()-1];
+      //auto& b = current_node.Blocks[i].instance.back();
+      PnRDB::block b = current_node.Blocks[i].instance[current_node.Blocks[i].instance.size()-1];
```

Comment:
The difference is the reference (&) not the auto.

Reply:
Do you have any example were using auto& core dumps? It is not core dumping on the switched capacitor filter with the mock PDK.

Reply:
This code looks wrong:
```c++
                      std::cout<<"core dump 1"<<std::endl;
                      for(unsigned int pin_index=0; pin_index <b.blockPins.size(); pin_index+=2){
                         pins.first = b.blockPins[pin_index].name;
                         pins.second = b.blockPins[pin_index+1].name;
                         pin_names.push_back(pins);
                      }
```
and least if the size of `b.blockPins` is not even. You should assert that it is.

Reply:
Seems that you can also use `const auto&` in this case and that might help find the problem.

Reply:
@stevenmburns 

Thanks.

Yaguang

Reply:
> This code looks wrong:
> 
> ```c++
>                       std::cout<<"core dump 1"<<std::endl;
>                       for(unsigned int pin_index=0; pin_index <b.blockPins.size(); pin_index+=2){
>                          pins.first = b.blockPins[pin_index].name;
>                          pins.second = b.blockPins[pin_index+1].name;
>                          pin_names.push_back(pins);
>                       }
> ```
> 
> and least if the size of `b.blockPins` is not even. You should assert that it is.

It's not a problem, every cap should have + pin and - pin.

Reply:
> Do you have any example were using auto& core dumps? It is not core dumping on the switched capacitor filter with the mock PDK.

When I run the case textcase_example, which is under PlacerRouterHierFlow, there is a core dump error.